RepositoryCompiler/CodeModel/CodeParsers/CSharp/CSharpCodeParser.cs

```diff
@@ -182,6 +184,9 @@ private List<CaDETClass> ConnectCaDETGraph(List<CaDETClass> classes)
             {
                 c.Parent = LinkParent(classes, c.Parent);
                 c.OuterClass = LinkOuterClass(classes, c.ContainerName);
+                c.FieldTypes = LinkFieldTypes(classes, c.Fields);
```

Comment:
You aren't really linking the types. You are generating the collections for the class, but the fields, variables, and return type of a method are not changed, meaning that our model is populated with dummy references that can't be used.

Once you remove these fields from the class the problem will present itself.